ui/src/pages/Profile.jsx

```diff
@@ -0,0 +1,53 @@
+import React, { useState, useEffect } from 'react'
+import { useQueryClient } from '@tanstack/react-query';
+import { Link, useNavigate } from "react-router-dom"
+import { getProfile } from '../api'
+
+function Profile() {
+  const [profile, setProfile] = useState(null)
+  const [loading, setLoading] = useState(true)
+  const [error, setError] = useState(null)
+  const navigate = useNavigate();
+  const queryClient = useQueryClient()
+
```

Comment:
Why you don't use the react query that way you not need the profile, loading  state in comes on the useQuery